recipes/benchmark/all/conanfile.py

```diff
@@ -49,20 +49,9 @@ def validate(self):
         if Version(self.version) < "1.7.0" and is_msvc(self) and self.options.shared:
             raise ConanInvalidConfiguration(f"{self.ref} doesn't support msvc shared builds")
 
-    def _cmake_new_enough(self, required_version):
-        try:
-            import re
-            from io import StringIO
-            output = StringIO()
-            self.run("cmake --version", output)
-            m = re.search(r'cmake version (\d+\.\d+\.\d+)', output.getvalue())
-            return Version(m.group(1)) >= required_version
-        except:
-            return False
-
     def build_requirements(self):
-        if Version(self.version) >= "1.7.1" and not self._cmake_new_enough("3.16.3"):
-            self.tool_requires("cmake/3.25.3")
+        if Version(self.version) >= "1.7.1":
+            self.tool_requires("cmake/[>=3.16 <4]")
```

Comment:
```suggestion
            self.tool_requires("cmake/[^3.16.3]")
```

Reply:
I think this way it's made more clear what the range allows. It can't just be me that I always have to double check the behaviour of ^ vs ~ when I see one in the wile

Reply:
At least add minor please.

Reply:
For future reference https://github.com/google/benchmark/commit/49aa374da96199d64fd3de9673b6f405bbc3de3e#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR1 is the important line for 1.7.1

Reply:
What I meant was to replace `[>=3.16 <4]` by `[>=3.16.3 <4]`